SnipperAPI/Controllers/SnippetsControllers.cs

```diff
@@ -26,14 +43,39 @@ public ActionResult<Snippet> CreateSnippet([FromBody] SnippetRequest snippetRequ
 
             // Returning the created snippet with Created status - use abstraction / restfulAPI
             return CreatedAtAction(nameof(GetSnippet), new { id = newSnippet.Id }, newSnippet);
+        }*/
+
+
+        [HttpPost("snippets")]
+        //[Authorize] // Requires authentication
+        public ActionResult<EncryptedSnippet> CreateSnippet([FromBody] SnippetRequest snippetRequest)
+        {
+            if (ModelState.IsValid)
+            {
+               // var currentUser = _userManager.GetUserAsync(User).Result;
+
+                // Encrypt the content before saving it
+                var encryptedContent =_encryptionService.Encrypt(snippetRequest.Content);
+
+                var newSnippet = new EncryptedSnippet(Id: Guid.NewGuid(), Title: snippetRequest.Title, Content: encryptedContent, CreatedAt: DateTime.UtcNow);
+                
+
+                Snippets.Add(newSnippet);
+
+                return CreatedAtAction(nameof(GetSnippet), new { id = newSnippet.Id }, newSnippet);
+            }
```

Comment:
Just typed up a suggestion to make the code more readable. When using the model constructors its nice just to new line each of the arguments as its a bit more readable than a long line of text.

```suggestion
            if (ModelState.IsValid)
            {
                var encryptedContent =_encryptionService.Encrypt(snippetRequest.Content);

                var newSnippet = new EncryptedSnippet(
                    Id: Guid.NewGuid(), 
                    Title: snippetRequest.Title, 
                    Content: encryptedContent, 
                    CreatedAt: DateTime.UtcNow);
                
                // Todo: Save snippet in some form of database, for now snippets are being stored in controller memory
                Snippets.Add(newSnippet);

                return CreatedAtAction(nameof(GetSnippet), new { id = newSnippet.Id }, newSnippet);
            }
```